array.h

```diff
@@ -2614,6 +2614,20 @@ void copy(const array<TSrc, ShapeSrc, AllocSrc>& src, array<TDst, ShapeDst, Allo
   copy(src.cref(), dst.ref());
 }
 
+/** Make a copy of the `src` array or array_ref with a new allocator `alloc`. */
+template <class T, class ShapeSrc,
+    class Alloc = std::allocator<typename std::remove_const<T>::type>>
+auto make_copy(const array_ref<T, ShapeSrc>& src, const Alloc& alloc = Alloc()) {
+  array<typename std::allocator_traits<Alloc>::value_type, ShapeSrc, Alloc> dst(src.shape(), alloc);
```

Comment:
Any reason this isn't just `T`?

Same for `make_move` below.

Reply:
One possibility: in the case of `array_ref`, `T` might be `const` or have other qualifiers(?), which we don't want when making a copy.